test/test_clients_api.js

```diff
@@ -75,11 +76,43 @@ describe('Clients API', () => {
             });
     });
 
-    it('should response 201 Created if request is valid', done => {
+    [{
+      reason: 'authRedirectUrls param is not an array of URLs',
+      body: { name: 'clientName', authRedirectUrls: 'notAnArrayOfUrls' }
+    }, {
+      reason: 'authRedirectUrls param is not an array of URLs',
+      body: { name: 'clientName',
+              authRedirectUrls: ['http://something.com'],
+              authFailureRedirectUrls: 'notAnArrayOfUrls' }
+    }, {
+      reason: 'authFailureRedirectUrls is present but authRedirectUrls is not',
+      body: { name: 'clientName',
+              authFailureRedirectUrls: ['http://something.com'] }
+    }].forEach(test => {
+      it('should respond 400 BadRequest if ' + test.reason, done => {
+        server.post(endpointPrefix + '/clients')
+              .set('Authorization', 'Bearer ' + token)
+              .send(test.body)
+              .expect('Content-type', /json/)
+              .expect(400)
+              .end((err, res) => {
+                res.status.should.be.equal(400);
+                res.body.code.should.be.equal(400);
+                res.body.errno.should.be.equal(
+                  errnos[ERRNO_INVALID_API_CLIENT_REDIRECT_URL]);
+                res.body.error.should.be.equal(errors[BAD_REQUEST]);
+                done();
+              });
+      });
+    });
+
+    it('should respond 201 Created if request is valid', done => {
       const name = 'clientName';
       server.post(endpointPrefix + '/clients')
             .set('Authorization', 'Bearer ' + token)
-            .send({ name: 'clientName' })
+            .send({ name: 'clientName',
+                    authRedirectUrls: ['http://something.com'],
+                    authFailureRedirectUrls: ['http://something.com'] })
```

Comment:
this should have found the issue; I don't know why this doesn't :/ (but locally it actually fails for me...)